### ISSUE TITLE
Problems deploying via CLI

### ISSUE THREAD

**Issue by scottcain** [lazerwalker/azure-mud#379]
I don't really want to turn you into Azure support for me, but I'm an Azure noob, so if you have anything obvious you can tell me, I'd appreciate it.

I installed the Azure CLI on my Ubuntu 20.04 machine and after logging in via a somewhat clumsy web interface, I did
```func  azure functionapp publish gmod-mud --typescript --force```
from inside the `server` directory. (I had to specify `typescript` because when I created the app I didn't know what platform to use, so I specified custom assuming I could come back and fix it when needed; the `--force` was to make that change happen).  It seems that this was not effective, and that the files presumably ended up in the wrong place.  When I go to the function app console and do an `ls` in `D:\home\site\wwwroot` I see the files that are in the `server` directory, so I must be at least close.  It seems to me that it might be possible that issuing that command one directory up might help, but I also don't want to make too much of a mess either.  Can you tell me if I'm on the right path?  Thanks!

**Comment by scottcain** [comment]
More information: logging in via twitter works (I get the twitter page that then redirects to my site, where I see a blank black page.  The js console tells me that `https://gmod-mud.azurewebsites.net/api/isRegistered` is returning a 404, which is why I went looking at the web-based console for my Azure function app, which showed the `isRegistered` directory and it's friends existing in `D:\home\site\wwwroot`.  Is there something I have to do to tell Azure to map incoming requests for `https://gmod-mud.azurewebsites.net/api/` to `D:\home\site\wwwroot`?

**Comment by lazerwalker** [comment]
Hm!

I'm not exactly sure how to fix why deployments don't seem to be working properly, but one easy way to check if they are is to go to the "Functions" tap in the Portal page for the Function App. If the deployment has worked properly, all of the API endpoints should show up there.

If you don't mind redoing the plumbing work of dropping in API keys, one way I've found to reliably configure the Function App correctly for function deploys is to create a new resource using VS Code. If you open the server folder in VS Code, you can use the Azure Functions plugin to deploy to an app, and if you let it create a new one for you then it should be configured correctly.

(FWIW, I'm also very close to having a template to automate employments entirely — the experience will basically be clicking a button, entering a project name, and you'll have a fully-working copy of the app with the latest code from my GitHub, the only manual step will be setting up Twitter/Google from their end)

**Comment by scottcain** [comment]
Yeah, the functions tab has "no results", so whatever happened, it didn't work via the CLI.  I will get VS Code; my place of comfort is remote linux machines and vim, but my co-worker says VS Code is "actually good" so I'll give it a shot. :-) 

Having a button will be nice--if you'd like me to try it out, I'm willing to wade through it.

**Comment by scottcain** [comment]
@lazerwalker Tiny steps forward.  I spent the afternoon getting VS Code installed and configured and successfully deployed to Azure (yay!), meaning that functions are available, and the Azure functions page says https://gmod-mud.azurewebsites.net/api/isRegistered is a valid function. Running from http://gmod-mud.scottcain.net/ however shows a cors failure (after somewhat of a delay, like Azure is thinking about it).  

I'm not sure what you meant by `If you don't mind redoing the plumbing work of dropping in API keys` because I didn't really have to do anything to get the deployment to work, but do I need to do something to get cors to work (I mean besides adding http://gmod-mud.scottcain.net to the allowed origins)?  Also tried restarting the function app after redeploying in case that was what was needed to no avail.

**Comment by lazerwalker** [comment]
What API call is triggering a CORS failure? My vague recollection is that Azure SignalR Service only works over HTTPS (unless running on localhost), so I wonder if running off of a non-SSL host is the issue.

When I mentioned dropping in API keys, I just meant that if you recreated the Function App from scratch there's more configuration to do (setting up CORS, authentication, and adding app settings to add connection data for Redis + SignalR)

**Comment by scottcain** [comment]
I'm getting this: ```Access to XMLHttpRequest at 'https://gmod-mud.azurewebsites.net/api/isRegistered' from origin 'http://gmod-mud.scottcain.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
networking.ts:433 Error Error: Network Error
    at module.exports (createError.js:16)
    at XMLHttpRequest.c.onerror (xhr.js:83)```

But I do have this:
<img width="553" alt="Screen Shot 2020-11-03 at 10 22 24 AM" src="https://user-images.githubusercontent.com/274751/98025439-83f81f00-1dbe-11eb-9871-617a9e1b9789.png">

Needing https will be a slight drag, but I imagine I can get one of those free certificates.  I suppose I can also run from localhost (now that I have a dev environment on this machine) as a test as well.

**Comment by scottcain** [comment]
Nope, I get the same CORS error when running from localhost as well: 

```
xhr.js:178 XHR finished loading: POST "https://gmod-mud.azurewebsites.net/.auth/me".
...
index.html:1 Access to XMLHttpRequest at 'https://gmod-mud.azurewebsites.net/api/isRegistered' from origin 'http://localhost:1234' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
networking.ts:433 Error Error: Network Error
    at createError (createError.js:16)
    at XMLHttpRequest.handleError (xhr.js:83)

```

**Comment by lazerwalker** [comment]
Based on your previous screenshot, I suspect the issue may be that you need to add http://localhost:1234 to the allowed CORS origins (both in the Function App and the SignalR resource).

That said, if you're interested in trying, I've just updated the README to include instructions for the new automated deployment flow, which should guarantee that the server setup is 100% correct.

**Comment by scottcain** [comment]
Ah, this piece is potentially important: `SignalR resource`.  I added localhost to the functions app but missed that I need to do it for SignalR as well.

I'll try the automated deployment later today.

**Comment by scottcain** [comment]
I started from scratch using the above mentioned automated deployment and that seems to have done the trick, and it now appears to be running successfully at http://gmod-mud.scottcain.net/ .  I have another issue (of course), but I'll open a new ticket for that.
